backend/dataall/modules/metadata_forms/services/metadata_form_enforcement_service.py

```diff
@@ -0,0 +1,341 @@
+                all_entities = session.query(entity_class)
+                if level == MetadataFormEnforcementScope.Organization:
+                    all_entities = all_entities.filter(
+                        entity_class.organizationUri.in_([org.organizationUri for org in orgs])
+                    )
+                if level == MetadataFormEnforcementScope.Environment:
+                    all_entities = all_entities.filter(
+                        entity_class.environmentUri.in_([env.environmentUri for env in envs])
+                    )
+                if level == MetadataFormEnforcementScope.Dataset:
+                    all_entities = all_entities.filter(entity_class.datasetUri.in_([ds.datasetUri for ds in datasets]))
+                all_entities = all_entities.all()
+                affected_entities.extend(
+                    [
+                        MetadataFormEnforcementService.form_affected_entity_object(entity_type, e, rule)
+                        for e in all_entities
+                    ]
+                )
+            return affected_entities
+
+    @staticmethod
+    def list_mf_enforcement_rules(uri):
+        with get_context().db_engine.scoped_session() as session:
+            return MetadataFormRepository.list_mf_enforcement_rules(session, uri)
+
+    @staticmethod
+    def paginate_mf_affected_entities(uri, data=None):
+        data = data or {}
+        return paginate_list(
+            items=MetadataFormEnforcementService.get_affected_entities(uri),
+            page=data.get('page', 1),
+            page_size=data.get('pageSize', 10),
+        ).to_dict()
+
+    @staticmethod
+    def resolve_home_entity(uri, rule=None):
+        with get_context().db_engine.scoped_session() as session:
+            if not rule:
+                rule = MetadataFormRepository.get_mf_enforcement_rule_by_uri(session, uri)
+            if rule.level == MetadataFormEnforcementScope.Global.value:
+                return ''
+            if rule.level == MetadataFormEnforcementScope.Organization.value:
+                return OrganizationRepository.get_organization_by_uri(session, rule.homeEntity).label
+            if rule.level == MetadataFormEnforcementScope.Environment.value:
+                return EnvironmentRepository.get_environment_by_uri(session, rule.homeEntity).label
+            if rule.level == MetadataFormEnforcementScope.Dataset.value:
+                return DatasetBaseRepository.get_dataset_by_uri(session, rule.homeEntity).label
+
+    @staticmethod
+    @TenantPolicyService.has_tenant_permission(MANAGE_METADATA_FORMS)
+    @MetadataFormAccessService.can_perform(ENFORCE_METADATA_FORM)
+    def delete_mf_enforcement_rule(uri, rule_uri):
+        with get_context().db_engine.scoped_session() as session:
+            rule = MetadataFormRepository.get_mf_enforcement_rule_by_uri(session, rule_uri)
+            session.delete(rule)
+            session.commit()
```

Comment:
this logic belongs to the repo, also first `get` then `delete` suboptimal, perhaps there is a native way to do it something like `session.query(MetadataFormEnforcementRule).filter_by(id=rule_uri).delete()`